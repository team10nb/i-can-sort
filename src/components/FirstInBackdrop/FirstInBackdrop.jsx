/*
    Author: Yijie Lu, Shiliang Chen
*/

import React from "react";
import Backdrop from "@material-ui/core/Backdrop";
import Card from "@material-ui/core/Card";
import CardContent from "@material-ui/core/CardContent";
import ButtonBase from "@material-ui/core/ButtonBase";
import Typography from "@material-ui/core/Typography";
import Snackbar from "@material-ui/core/Snackbar";
import Alert from "@material-ui/lab/Alert";
import TutorialMainPage from '../../scenes/mainPages/Tutorial';
import { makeStyles } from "@material-ui/core/styles";
import { Link } from "react-router-dom";
import { motion } from "framer-motion";

const useStyles = makeStyles((theme) => ({
    backdrop: {
        zIndex: theme.zIndex.drawer + 1,
        color: "#212121",
    },
    cardBackground: {
        width: 650,
        height: 460,
        // backgroundColor: "white",
        backgroundColor: "rgba(255,255,255,0.8)",
        borderRadius: "20px",
    },
    buttonBase: {
        display: "flex",
        "& > * + *": {
            marginLeft: theme.spacing(8),
        },
        justifyContent: "center",
    },
    cardOne: {
        opacity: 1,
        width: 230,
        height: 230,
        backgroundColor: "#4caf50",
        color: "white",
        borderRadius: "30px",
        "&:hover, &$focusVisible": {
            transform: "scale(1.05, 1.05)",
            // boxShadow: "0 0 10 5 #888888,",
        },
        "&:active, &$focusVisible": {
            transform: "scale(1, 1 )",
        },
        transition: "0.5s",
    },
    cardTwo: {
        width: 230,
        height: 230,
        backgroundColor: "#1565c0",
        color: "white",
        borderRadius: "30px",
        "&:hover, &$focusVisible": {
            transform: "scale(1.05, 1.05)",
        },
        "&:active, &$focusVisible": {
            transform: "scale(1, 1 )",
        },
        transition: "0.5s",
    },
    title: {
        marginTop: "33px",
    },
    tip: {
        marginLeft: 52,
        marginRight: 50,
        marginBottom: 23,
        justifyContent: "center",
        textAlign: "center",
    },
    content1: {
        marginTop: "50px",
        borderRadius: "30px",
    },
    content2: {
        marginTop: "52px",
        borderRadius: "30px",
    },
    icon: {
        color: "white",
        fontSize: "3rem",
    },
}));

export default function FirstInBackdrop(props) {
    const classes = useStyles();
    const snack = localStorage.getItem("snack")
        ? JSON.parse(localStorage.getItem("snack"))
        : null;
    const [open, setOpen] = React.useState(true);
    const [appear, setAppear] = React.useState(true);
    const handleClose = (event, reason) => {
        if (reason === "clickaway") {
            return;
        }
        localStorage.setItem("snack", JSON.stringify(0));
        setAppear(false);
    };
    const onClick1 = () => {
        localStorage.setItem("first", JSON.stringify(0));
        window.location.href = "/TutorialMainPage";
    };
    const onClick = localStorage.setItem("first", JSON.stringify(0));
    return (
        <div>
            <Backdrop className={classes.backdrop} open={open}>
                <motion.div
                    initial={{ opacity: 0.2, scale: 0 }}
                    animate={{
                        opacity: 1,
                        scale: 1,
                    }}
                    transition={{
                        type: "spring",
                        stiffness: 150,
                        damping: 15,
                        mass: 0.1,
                    }}
                >
                    <Card className={classes.cardBackground}>
                        <div>
                            <Typography
                                variant="h4"
                                gutterBottom
                                style={{
                                    fontFamily: "Quicksand",
                                    fontWeight: "600",
                                    color: "#0e4686",
                                }}
                                className={classes.title}
                            >
                                Welcome!
                            </Typography>
                            <Typography
                                gutterBottom
                                style={{
                                    fontFamily: "Quicksand",
                                    fontWeight: "600",
                                    color: "#0e4686",
                                }}
                                className={classes.tip}
                            >
                                We designed a special section of Tutorial for
                                freshmen. <br />
                                You can learn sorting and correctness in other
                                two sections. <br />
                                Do you have any experience on programming?{" "}
                                <br />
                            </Typography>
                        </div>

                        <div className={classes.buttonBase}>
                            <ButtonBase style={{ borderRadius: "30px" }}>
                                <Link
                                    to="/TutorialMainPage"
                                    style={{ textDecoration: "none" }}
                                >
                                    <motion.div
                                        initial={{ y: -50, opacity: 0 }}
                                        animate={{
                                            y: 0,
                                            opacity: 1,
                                        }}
                                        transition={{
                                            delay: 0.4,
                                        }}
                                    >
                                        <Card className={classes.cardOne}>
                                            <CardContent
                                                className={classes.content1}
                                            >
                                                <Typography
                                                    variant="h6"
                                                    gutterBottom
                                                    style={{
                                                        fontFamily: "inherit",
                                                        fontWeight: "600",
                                                    }}
                                                >
                                                    I'm new at programming, go
                                                    to Tutorial first!
                                                </Typography>
                                            </CardContent>
                                        </Card>
                                    </motion.div>
                                </Link>
                            </ButtonBase>
                            <ButtonBase style={{ borderRadius: "30px" }}>
                                <Link
                                    to="./ProcedureMainPage"
                                    style={{ textDecoration: "none" }}
                                >
                                    <motion.div
                                        initial={{ y: -50, opacity: 0 }}
                                        animate={{
                                            y: 0,
                                            opacity: 1,
                                        }}
                                        transition={{
                                            delay: 0.6,
                                        }}
                                    >
                                        <Card className={classes.cardTwo}>
                                            <CardContent
                                                className={classes.content2}
                                            >
                                                <Typography
                                                    variant="h6"
                                                    gutterBottom
                                                    style={{
                                                        fontFamily: "inherit",
                                                        fontWeight: "600",
                                                    }}
                                                >
                                                    I've got basic knowledge,
                                                    start now!
                                                </Typography>
                                            </CardContent>
                                        </Card>
                                    </motion.div>
                                </Link>
                            </ButtonBase>
                        </div>
                    </Card>
                </motion.div>
                {snack === 1 ? (
                    <Snackbar open={appear} onClose={handleClose} style={{bottom:"80px"}}>
                        <Alert onClose={handleClose} severity="success">
                            Successfully reset history!
                        </Alert>
                    </Snackbar>
                ) : (
                    <div />
                )}
            </Backdrop>
        </div>
    );
}
