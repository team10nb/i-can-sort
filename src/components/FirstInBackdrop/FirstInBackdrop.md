
### **Developer**

Shiliang Chen, Yijie Lu

Reference: 

MATERIAL-UI (https://material-ui.com)

Framer Motion (https://framer.com/motion)

###  **Introduction**

A popup which asks users whether a beginner or have knowledge, and jump to Tutorial page or Procedure page.

###  **Import**

```html
import React from 'react';
import Backdrop from '@material-ui/core/Backdrop';
import Card from '@material-ui/core/Card';
import CardContent from '@material-ui/core/CardContent';
import ButtonBase from "@material-ui/core/ButtonBase";
import Typography from '@material-ui/core/Typography';
import { makeStyles } from '@material-ui/core/styles';
import {Link} from 'react-router-dom';
import { motion } from "framer-motion";
```

###  **Props**


| Name | Type | Default | Description |
| ---- | ---- | ------- | ----------- |
| / | / | / | / |


###  **Methods**

| Name | Props | Return | Description |
| ---- | ----- | ------ | ----------- |
|/|/|    /     |     /      |


###  **Demo**

```jsx
< FirstInBackdrop
/>
```
