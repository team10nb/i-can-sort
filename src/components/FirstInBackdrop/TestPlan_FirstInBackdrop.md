## Test Plan 
--------
<br>

#### ComponentName: FirstInBackdrop

|Test Name|Description|Operation|Expected Outcome|
|----|------|--------|----------------|
| should match FirstInBackdrop Snapshot | Compare new snapshot with old one. | null | Snapshots are the same. |








