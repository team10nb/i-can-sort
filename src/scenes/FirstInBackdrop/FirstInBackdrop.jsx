import React from 'react';
import Backdrop from '@material-ui/core/Backdrop';
import Card from '@material-ui/core/Card';
import CardActions from '@material-ui/core/CardActions';
import CardContent from '@material-ui/core/CardContent';
import ButtonBase from "@material-ui/core/ButtonBase";
import Typography from '@material-ui/core/Typography';
import { makeStyles } from '@material-ui/core/styles';
import ChildCareIcon from '@material-ui/icons/ChildCare';
import FaceIcon from '@material-ui/icons/Face';
import {Link} from 'react-router-dom';

const useStyles = makeStyles((theme) => ({
  backdrop: {
    zIndex: theme.zIndex.drawer + 1,
    color: "#212121",
  },
  cardOne: {
    display: "flex",
        // '& > *': {
        //     marginLe: theme.spacing(4),
        //   },
        '& > * + *': {     
            marginLeft: theme.spacing(8),
        },
        justifyContent: "center",
    width: 650,
    height: 400,
    backgroundColor: "white",
    opacity:0.8,
    borderRadius:"20px",
  },
  cardTwo:{
    width: 230,
    height: 230,
    backgroundColor:"#1565c0",
    color:"white",
    borderRadius:"30px", 
  },
  content1:{
    marginTop:"69px"
  },
  content2:{
    marginTop: "47px",       
  },
  icon:{
    color:"white",
    fontSize :"3rem"
  }
}));


export default function FirstInBackdrop(props) {
  const classes = useStyles();
  const [open, setOpen] = React.useState(true);
  const onClick1 = () => {
    //return to tutorial main page 
    localStorage.setItem("first", JSON.stringify(0));
    window.location.href = "/TutorialMainPage";
  };
  // const onClick2 = () => {
  //   //return to procedure main page 
  //   localStorage.setItem("first", JSON.stringify(0));
  //   window.location.href = "/ProcedureMainPage";
  // };
  const onClick = localStorage.setItem("first", JSON.stringify(0));
  return (
    <div>
      <Backdrop className={classes.backdrop} open={open} >
        <Card className = {classes.cardOne}>
        
          <ButtonBase >
            <Link to="./TutorialMainPage" style={{textDecoration:"none"}}>
                       
            <Card className = {classes.cardTwo} >
              <CardContent className = {classes.content1}>
                <Typography variant="h6" gutterBottom>I am a Freshman, go to tutorial first!
                </Typography>
              </CardContent>
            </Card>
            </Link>
          </ButtonBase>
          
          
          <ButtonBase >
            <Link to="./ProcedureMainPage" style={{textDecoration:"none"}}>
            
            <Card className = {classes.cardTwo} >         
              <CardContent className = {classes.content2}>
              <Typography variant="h6" gutterBottom>I have already know the basic knowledge, quickly start now!
              </Typography>
              </CardContent>
            </Card>
            </Link>
          </ButtonBase>
          
        </Card>
      </Backdrop>
    </div>
  );
}




