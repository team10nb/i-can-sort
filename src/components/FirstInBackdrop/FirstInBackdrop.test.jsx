/*
    Author: Shiliang Chen, Yijie Lu
*/
import FirstInBackdrop from "./FirstInBackdrop";
import { render } from "@testing-library/react";

jest.mock("react-router-dom",()=>{
    return {
        Link: () => <div>mockLink</div>,

    }
})

// test( 'should contain buttons', () => {
//     const content = render(<FirstInBackdrop  />)
//     const {getByText, getAllByText, getAllByRole} = content;
//     getByText("I'm new at this, go to Tutorial first!");
//     getByText("I've got' basic knowledge, start now!");




// })

test( 'should match FirstInBackdrop snapshot', () => {
    const content = render(<FirstInBackdrop   />);
    expect(content).toMatchSnapshot();

})
