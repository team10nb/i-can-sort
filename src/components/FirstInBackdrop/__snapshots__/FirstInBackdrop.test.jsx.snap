// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should match FirstInBackdrop snapshot 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div>
        <div
          aria-hidden="true"
          class="MuiBackdrop-root makeStyles-backdrop-1"
          style="opacity: 1; webkit-transition: opacity 225ms cubic-bezier(0.4, 0, 0.2, 1) 0ms; transition: opacity 225ms cubic-bezier(0.4, 0, 0.2, 1) 0ms;"
        >
          <div
            style="opacity: 0.2; transform: scale(0) translateZ(0);"
          >
            <div
              class="MuiPaper-root MuiCard-root makeStyles-cardBackground-2 MuiPaper-elevation1 MuiPaper-rounded"
            >
              <div>
                <h4
                  class="MuiTypography-root makeStyles-title-6 MuiTypography-h4 MuiTypography-gutterBottom"
                  style="font-family: Quicksand; font-weight: 600; color: rgb(14, 70, 134);"
                >
                  Welcome!
                </h4>
                <p
                  class="MuiTypography-root makeStyles-tip-7 MuiTypography-body1 MuiTypography-gutterBottom"
                  style="font-family: Quicksand; font-weight: 600; color: rgb(14, 70, 134);"
                >
                  We designed a special section of Tutorial for freshmen. 
                  <br />
                  Do you have any experience on programming? 
                  <br />
                </p>
              </div>
              <div
                class="makeStyles-buttonBase-3"
              >
                <button
                  class="MuiButtonBase-root"
                  tabindex="0"
                  type="button"
                >
                  <div>
                    mockLink
                  </div>
                  <span
                    class="MuiTouchRipple-root"
                  />
                </button>
                <button
                  class="MuiButtonBase-root"
                  tabindex="0"
                  type="button"
                >
                  <div>
                    mockLink
                  </div>
                  <span
                    class="MuiTouchRipple-root"
                  />
                </button>
              </div>
            </div>
          </div>
        </div>
      </div>
    </div>
  </body>,
  "container": <div>
    <div>
      <div
        aria-hidden="true"
        class="MuiBackdrop-root makeStyles-backdrop-1"
        style="opacity: 1; webkit-transition: opacity 225ms cubic-bezier(0.4, 0, 0.2, 1) 0ms; transition: opacity 225ms cubic-bezier(0.4, 0, 0.2, 1) 0ms;"
      >
        <div
          style="opacity: 0.2; transform: scale(0) translateZ(0);"
        >
          <div
            class="MuiPaper-root MuiCard-root makeStyles-cardBackground-2 MuiPaper-elevation1 MuiPaper-rounded"
          >
            <div>
              <h4
                class="MuiTypography-root makeStyles-title-6 MuiTypography-h4 MuiTypography-gutterBottom"
                style="font-family: Quicksand; font-weight: 600; color: rgb(14, 70, 134);"
              >
                Welcome!
              </h4>
              <p
                class="MuiTypography-root makeStyles-tip-7 MuiTypography-body1 MuiTypography-gutterBottom"
                style="font-family: Quicksand; font-weight: 600; color: rgb(14, 70, 134);"
              >
                We designed a special section of Tutorial for freshmen. 
                <br />
                Do you have any experience on programming? 
                <br />
              </p>
            </div>
            <div
              class="makeStyles-buttonBase-3"
            >
              <button
                class="MuiButtonBase-root"
                tabindex="0"
                type="button"
              >
                <div>
                  mockLink
                </div>
                <span
                  class="MuiTouchRipple-root"
                />
              </button>
              <button
                class="MuiButtonBase-root"
                tabindex="0"
                type="button"
              >
                <div>
                  mockLink
                </div>
                <span
                  class="MuiTouchRipple-root"
                />
              </button>
            </div>
          </div>
        </div>
      </div>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
